pkg/types/var.go

```diff
@@ -125,11 +126,16 @@ func (vs *VarSet) MergeSlice(incoming []Var) error {
 // Merge absorbs another Var with error on name collision.
 // Empty fields in incoming Var is defaulted.
 func (vs *VarSet) Merge(v Var) error {
+	v.defaulting()
 	if vs.Contains(v) {
-		return fmt.Errorf(
-			"var '%s' already encountered", v.Name)
+		// Only return an error if a variable with the same
+		// name already exists
+		if !reflect.DeepEqual(v, vs.set[v.Name]) {
+			return fmt.Errorf(
+				"var '%s' already encountered", v.Name)
```

Comment:
I think this is what `vs.Contains` does.

Reply:
No vs.Contains is using the name of the variable. The DeepEquals allows to detect that we are trying to insert a new variable with the same name but a different definition.